File: javascript/easy/findAllNumbersDisappearedInAnArray.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findDisappearedNumbers = function(nums) {
    const res = new Array();
    for(let num of nums) {
        let index = Math.abs(num) - 1;
        if(nums[index] > 0) {
            nums[index] = -nums[index];
        }
    }
    
    for(let i = 0; i < nums.length; i++) {
        if(nums[i] > 0) {
            res.push(i+1);
        }
    }
    return res;
};

File: ruby/easy/find_all_numbers_disappeared_in_an_array.rb
# @param {Integer[]} nums
# @return {Integer[]}
def find_disappeared_numbers(nums)
    res = Array.new
    nums.each do |num|
        i = num.abs
        nums[i-1] = -nums[i-1] if nums[i-1] > 0
    end
    for i in 0..nums.length-1
        res.push(i+1) if nums[i] > 0
    end
    return res
end
